### PR TITLE
feat: add nemo-response-guard plugin for output rail checks

### PR DESCRIPTION
**Pull request by liavweiss**
### Summary

Adds a `nemo-response-guard` BBR plugin that implements `ResponseProcessor` to check model output against NeMo Guardrails output rails before returning responses to the caller.

### Changes

- **`pkg/plugins/nemo/response_guard.go`** — `NemoResponseGuardPlugin` implementing `framework.ResponseProcessor`. Extracts assistant messages from `choices[].message.content`, POSTs them to NeMo's `/v1/guardrail/checks`, and returns `403 Forbidden` if NeMo blocks the content.
- **`pkg/plugins/nemo/response_guard_test.go`** — Full test coverage: 14 `ProcessResponse` scenarios (allow, block, NeMo errors, no-ops, fail-closed), 11 `extractAssistantMessages` unit tests, factory tests, payload verification, model forwarding, trailing slash, and nil response handling.
- **`cmd/main.go`** — Registers `NemoResponseGuardFactory`.

### Verified (E2E on Kind cluster)

**Safe response — passes both guards (200 OK):**

```text
POST /v1/chat/completions → 200 OK
{"choices":[{"message":{"role":"assistant","content":"Hello! How can I help you?"}}]}

Logs: nemo-request-guard → Allowed → nemo-response-guard → Allowed
```
**Toxic response — blocked by response guard (403 Forbidden):**

```
POST /v1/chat/completions → 403 Forbidden
inference error: Forbidden - response blocked by NeMo guardrails

Logs: nemo-request-guard → Allowed → nemo-response-guard → Blocked by NeMo guardrails (check content safety output: blocked)
```

## Summary by CodeRabbit

* **New Features**
  * Added a NeMo response-guard plugin that can be enabled and configured (including timeout settings); it is now registered at application startup.

* **Tests**
  * Added comprehensive test coverage for the response guard, covering configuration validation, various response scenarios, error handling, and edge cases.

### CODE REVIEW

**Review by nirrozenbaum — COMMENTED** [review]
can you please add TypedName as done in #169 ?

File: pkg/plugins/nemo/response_guard.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+// if NeMo flags the content.
+//
+// NeMo always returns HTTP 200 for both allowed and blocked responses. The block/allow
+// decision is conveyed through the response body "status" field.
+// "success" means the response passed all rails; "blocked" means it is blocked.
+func (p *NemoResponseGuardPlugin) ProcessResponse(ctx context.Context, _ *framework.CycleState, response *framework.InferenceResponse) error {
+	if response == nil || response.Body == nil {
+		return nil
+	}
+
```
Comment (nirrozenbaum) [inline pkg/plugins/nemo/response_guard.go]: can you remove this?
response and body can be assumed not to be nil assumed (handled by the framework).

```suggestion
```

File: pkg/plugins/nemo/response_guard.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		TimeoutSeconds: defaultTimeoutSec,
+	}
+
+	if len(rawParameters) > 0 {
+		if err := json.Unmarshal(rawParameters, &config); err != nil {
+			return nil, fmt.Errorf("failed to parse the parameters of the '%s' plugin - %w", NemoResponseGuardPluginType, err)
+		}
+	}
+
+	p, err := NewNemoResponseGuardPlugin(config.NemoURL, config.TimeoutSeconds)
```
Comment (nirrozenbaum) [inline pkg/plugins/nemo/response_guard.go]: nit: rename `p` to `plugin` (just to be consistent with the other places)

File: pkg/plugins/nemo/response_guard.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+// NemoResponseGuardFactory is the factory function for NemoResponseGuardPlugin.
+func NemoResponseGuardFactory(name string, rawParameters json.RawMessage, _ framework.Handle) (framework.BBRPlugin, error) {
+	config := nemoGuardConfig{
+		TimeoutSeconds: defaultTimeoutSec,
+	}
+
+	if len(rawParameters) > 0 {
+		if err := json.Unmarshal(rawParameters, &config); err != nil {
+			return nil, fmt.Errorf("failed to parse the parameters of the '%s' plugin - %w", NemoResponseGuardPluginType, err)
```
Comment (nirrozenbaum) [inline pkg/plugins/nemo/response_guard.go]: super nit (consistent with request plugin):
```suggestion
			return nil, fmt.Errorf("failed to parse the parameters of '%s' plugin - %w", NemoResponseGuardPluginType, err)
```

File: pkg/plugins/nemo/response_guard_test.go [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+			serverHandler: func(w http.ResponseWriter, r *http.Request) {
+				if err := json.NewEncoder(w).Encode(map[string]any{"status": "allowed"}); err != nil {
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+				}
+			},
+			body:            validResponseBody("Hello"),
+			wantErr:         true,
+			wantErrContains: forbiddenMsg,
+			wantErrCode:     errcommon.Forbidden,
+		},
```
Comment (nirrozenbaum) [inline pkg/plugins/nemo/response_guard_test.go]: we can probably remove this test.

```suggestion

```

File: pkg/plugins/nemo/response_guard_test.go [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+				if err := json.NewEncoder(w).Encode(map[string]any{
+					"status": "blocked",
+					"rails_status": map[string]any{
+						`huggingface detector check output $hf_model="ibm-granite/granite-guardian-hap-38m"`: map[string]any{"status": "blocked"},
+					},
+				}); err != nil {
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+				}
+			},
+			body:            validResponseBody("You stupid moron."),
```
Comment (nirrozenbaum) [inline pkg/plugins/nemo/response_guard_test.go]: since we use mock to mimic guardrails blocking a request, we don't really need to use inappropriate messages :). 
maybe we can change to a text that is still clear that should be block, e.g.,:

```suggestion
			body:            validResponseBody("this message should be blocked."),
```

File: pkg/plugins/nemo/response_guard_test.go [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+// --- nil response ---
+
+func TestNemoResponseGuardNilResponse(t *testing.T) {
+	p, err := NewNemoResponseGuardPlugin("http://nemo:8000/v1/guardrail/checks", 30)
+	require.NoError(t, err)
+
+	err = p.ProcessResponse(context.Background(), framework.NewCycleState(), nil)
+	assert.NoError(t, err, "nil response should be a no-op")
+}
+
```
Comment (nirrozenbaum) [inline pkg/plugins/nemo/response_guard_test.go]: can be removed:

```suggestion
```

**Review by nirrozenbaum — COMMENTED** [review]
@liavweiss nice, looks good. 
left some minor comments.

**Comment by liavweiss** [comment]
> @liavweiss nice, looks good. left some minor comments.

@nirrozenbaum, thanks for the comments, pr is ready.

**Review by nirrozenbaum — APPROVED** [review]
/lgtm
/approve

Thanks! 🙏
